aptos-indexer-processors-sdk/testing-framework/src/new_test_context.rs

```diff
@@ -0,0 +1,196 @@
+use std::fs;
+use std::path::Path;
+use aptos_protos::indexer::v1::TransactionsResponse;
+use aptos_indexer_processor_sdk::traits::processor_trait::ProcessorTrait;
+use aptos_protos::transaction::v1::Transaction;
+use async_trait::async_trait;
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    ContainerAsync, GenericImage, ImageExt,
+};
+use url::Url;
+use aptos_indexer_processor_sdk::aptos_indexer_transaction_stream::TransactionStreamConfig;
+use crate::mock_grpc::MockGrpcServer;
+
+pub struct SdkTestContext<D: TestDatabase> {
+    pub transaction_batches: Vec<Transaction>,
+    pub database: D, // Holds the database setup (user-defined)
```

Comment:
QQ about this struct? Why does it need to store the database? I wonder if it could be moved out of this struct and managed as its own struct? That would allow more flexible database setups.